### PR TITLE
tests: Add k8s negative policy test

### LINKED ISSUE #9437: tests: add k8s negative policy test

### PR DESCRIPTION
**Pull request by GabyCT**
This PR adds a k8s negative policy test to the confidential attestation bats test.

Fixes #9437

### CODE REVIEW

**Comment by GabyCT** [comment]
When testing with `kbs_set_allow_all_resources`, we can see the following
```
$ kubectl logs aa-test-cc
% Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
aatest
100     7  100     7    0     0     29      0 --:--:-- --:--:-- --:--:--    29
```
And when testing with `kbs_set_deny_all_resources` we can see the following
```
$ kubectl logs aa-test-cc
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   319  100   319    0     0    312      0  0:00:01  0:00:01 --:--:--   312
```
No major info is being provided it

**Comment by fitzthum** [comment]
> No major info is being provided it.

Hmm. Not sure why we don't see some kind of output on error. @Xynnn007 

I wouldn't expect to find any difference with `kubectl describe` or any of the other kube apis.

**Comment by Xynnn007** [comment]
It is weird. Could you show or where can I get the logs of the KBS/AS side?

File: tests/integration/kubernetes/k8s-confidential-attestation.bats [hunk 1]
```diff
@@ -72,6 +72,36 @@ setup() {
 	kubectl logs aa-test-cc | grep -q "aatest"
 }
 
+@test "Negative policy test" {
```
Comment (wainersm) [inline tests/integration/kubernetes/k8s-confidential-attestation.bats:75]: Nit: I suggest a new test name "Cannot get CDH resource when deny-all policy is set"

File: tests/integration/kubernetes/k8s-confidential-attestation.bats [hunk 2]
```diff
@@ -72,6 +72,36 @@ setup() {
...
+
+	kubectl apply -f "${K8S_TEST_YAML}"
+
+	# Retrieve pod name, wait for it to come up, retrieve pod ip
+	export pod_name=$(kubectl get pod -o wide | grep "aa-test-cc" | awk '{print $1;}')
+
+	# Check pod creation
+	kubectl wait --for=condition=Ready --timeout="$timeout" pod "${pod_name}"
+
+	sleep 5
```
Comment (wainersm) [inline tests/integration/kubernetes/k8s-confidential-attestation.bats:99]: After the sleep, could you run `kubectl logs aa-test-cc` to help on debugging?

**Comment by GabyCT** [comment]
@wainersm changes applied

**Comment by fitzthum** [comment]
I think we are going to be making some changes to error handling in the ASR and CDH, so we can probably separate that from this PR and skip checking for a particular error response for now.

**Comment by wainersm** [comment]
> When testing with `kbs_set_allow_all_resources`, we can see the following
> 
> ```
> $ kubectl logs aa-test-cc
> % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
>                                  Dload  Upload   Total   Spent    Left  Speed
> aatest
> 100     7  100     7    0     0     29      0 --:--:-- --:--:-- --:--:--    29
> ```
> 
> And when testing with `kbs_set_deny_all_resources` we can see the following
> 
> ```
> $ kubectl logs aa-test-cc
>   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
>                                  Dload  Upload   Total   Spent    Left  Speed
> 100   319  100   319    0     0    312      0  0:00:01  0:00:01 --:--:--   312
> ```
> 
> No major info is being provided it. I also try to see if I could spot some difference when running with `kubectl describe pods` but the info is the same for both cases.

Hi @GabyCT !

I don't know if I fully understood your concerns.

I expect  `kubectl describe pods` output be the same. The pod doesn't fail because it shouldn't even when the secret cannot be fetched (see the pod https://github.com/kata-containers/kata-containers/blob/main/tests/integration/kubernetes/runtimeclass_workloads/pod-attestable.yaml).

When `kbs_set_deny_all_resources` the out of `curl` is empty indeed. As @fitzthum already said, they are working on logging error messages so then we will be able to grep for specific errors. Not now, not in this PR :D

**Comment by GabyCT** [comment]
> > When testing with `kbs_set_allow_all_resources`, we can see the following
> > ```
> > $ kubectl logs aa-test-cc
> > % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
> >                                  Dload  Upload   Total   Spent    Left  Speed
> > aatest
> > 100     7  100     7    0     0     29      0 --:--:-- --:--:-- --:--:--    29
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > And when testing with `kbs_set_deny_all_resources` we can see the following
> > ```
> > $ kubectl logs aa-test-cc
> >   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
> >                                  Dload  Upload   Total   Spent    Left  Speed
> > 100   319  100   319    0     0    312      0  0:00:01  0:00:01 --:--:--   312
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > No major info is being provided it. I also try to see if I could spot some difference when running with `kubectl describe pods` but the info is the same for both cases.
> 
> Hi @GabyCT !
> 
> I don't know if I fully understood your concerns.
> 
> I expect `kubectl describe pods` output be the same. The pod doesn't fail because it shouldn't even when the secret cannot be fetched (see the pod https://github.com/kata-containers/kata-containers/blob/main/tests/integration/kubernetes/runtimeclass_workloads/pod-attestable.yaml).
> 
> When `kbs_set_deny_all_resources` the out of `curl` is empty indeed. As @fitzthum already said, they are working on logging error messages so then we will be able to grep for specific errors. Not now, not in this PR :D



> > When testing with `kbs_set_allow_all_resources`, we can see the following
> > ```
> > $ kubectl logs aa-test-cc
> > % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
> >                                  Dload  Upload   Total   Spent    Left  Speed
> > aatest
> > 100     7  100     7    0     0     29      0 --:--:-- --:--:-- --:--:--    29
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > And when testing with `kbs_set_deny_all_resources` we can see the following
> > ```
> > $ kubectl logs aa-test-cc
> >   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
> >                                  Dload  Upload   Total   Spent    Left  Speed
> > 100   319  100   319    0     0    312      0  0:00:01  0:00:01 --:--:--   312
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > No major info is being provided it. I also try to see if I could spot some difference when running with `kubectl describe pods` but the info is the same for both cases.
> 
> Hi @GabyCT !
> 
> I don't know if I fully understood your concerns.
> 
> I expect `kubectl describe pods` output be the same. The pod doesn't fail because it shouldn't even when the secret cannot be fetched (see the pod https://github.com/kata-containers/kata-containers/blob/main/tests/integration/kubernetes/runtimeclass_workloads/pod-attestable.yaml).
> 
> When `kbs_set_deny_all_resources` the out of `curl` is empty indeed. As @fitzthum already said, they are working on logging error messages so then we will be able to grep for specific errors. Not now, not in this PR :D

@wainer, no I was just putting that the description is the same, I do not have a concern there

**Comment by GabyCT** [comment]
@wainersm changes applied

**Review by stevenhorsman — APPROVED** [review]
The code looks good to me (based on the agreement that enhancing it to check for an error in the logs will come as a separate PR) and both tests pass on the non-TEE job. Thanks @GabyCT and co!

**Comment by fitzthum** [comment]
There have been a few changes to error handling, so it's possible that bumping guest-components to the latest version would give you an error to check on failure. We can do that in a separate PR tho. Up to you @GabyCT

**Comment by GabyCT** [comment]
> There have been a few changes to error handling, so it's possible that bumping guest-components to the latest version would give you an error to check on failure. We can do that in a separate PR tho. Up to you @GabyCT

@fitzthum thanks for the info, yes I can make a separate PR to check that we can see an error check

**Comment by wainersm** [comment]
> The code looks good to me (based on the agreement that enhancing it to check for an error in the logs will come as a separate PR) and both tests pass on the non-TEE job. Thanks @GabyCT and co!

They are passing! \o/

```
ok 2 Cannot get CDH resource when deny-all policy is set
```

**Review by wainersm — APPROVED** [review]
LGTM. Thanks @GabyCT !

**Pull request merged**